model-optimizer/extensions/middle/ReverseV2ToReverseSequence.py

```diff
@@ -32,29 +38,55 @@ def pattern():
 
     def replace_pattern(self, graph: Graph, match: dict):
         reverse = match['reverse']
-        input_data_shape = reverse.in_node(0).shape
-
-        if len(input_data_shape) == 1:
-            raise Error('Reverse operation name = {} is\'t supported because of 1D input.'.format(reverse.name))
+        reverse_name = reverse.soft_get('name', reverse.id)
 
         assert reverse.in_port(1).disconnected()
 
-        seq_axis = reverse['axis']
-        # We need to choose arbitrary batch_axis != sequence_axis
-        batch_axis = int(not seq_axis)
+        # add new dimension as batch
+        unsq_node = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
+                                                     {'name': reverse_name+"/Unsqueeze"})
+        reverse.in_port(0).get_source().connect(unsq_node.in_port(0))
+        # set layout as correct to avoid adding Transpose nodes
+        unsq_node.in_port(1).__setattr__('input_permutation', None)
```

Comment:
Now we seem to add `Unsqueeze` and `Squeeze` for all cases including `rank>1`. I propose to differentiate these cases and handle it separately. It can lead to performance drop.

Reply:
The other way is to stay rank=1 unimplemented and add `TODO`. I presume due to these Squeeze and Unsqueeze you have to add new transformation for TI with ReverseSequence

Reply:
fixed